### PR TITLE
feat(appengine): Enable new artifacts for config artifacts in deploy SG

### PR DESCRIPTION
**Pull request by Jammy-Louie**
enable the new expected artifacts for config artifacts when deploying a new server group.

### CODE REVIEW

**Comment by Jammy-Louie** [comment]
corresponding PR:
https://github.com/spinnaker/orca/pull/3013

this is to update the config artifacts section of deploy server group

rewrite:
Screen Shot 2019-06-26 at 2 45 25 PM

Pre rewrite:
Screen Shot 2019-06-26 at 2 44 07 PM

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  };
+
+  return (
+    <>
+      {props.configArtifacts.map((a, i) => {
+        return (
+          <div
+            key={i}
+            className={
+              'artifact-configuration-section col-md-12' + (props.configArtifacts.length - 1 === i ? 'last-entry' : '')
```
Comment (maggieneterval) [inline app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]: maybe cleaner to use [classnames](https://github.com/JedWatson/classnames) here?

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    newConfigArtifacts.splice(index, 1, { ...newConfigArtifacts[index], account });
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  return (
+    <>
+      {props.configArtifacts.map((a, i) => {
+        return (
+          <div
+            key={i}
```
Comment (maggieneterval) [inline app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]:
```suggestion
            key={a.id}
```

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            className={
+              'artifact-configuration-section col-md-12' + (props.configArtifacts.length - 1 === i ? 'last-entry' : '')
+            }
+          >
+            <div className={'col-md-9'}>
```
Comment (maggieneterval) [inline app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]:
```suggestion
            <div className="col-md-9">
```

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  stage: IStage;
+  updateConfigArtifacts: (configArtifacts: any[]) => void;
+  updatePipeline: (changes: Partial<IPipeline>) => void;
+}
+
+export const ConfigFileArtifactList = (props: IConfigFileArtifactListProps) => {
+  const addConfigArtifact = () => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.push({});
+    props.updateConfigArtifacts(newConfigArtifacts);
```
Comment (maggieneterval) [inline app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]: feel free to ignore this suggestion but could probably do this in one line with `concat`, which returns a new array
```suggestion
    props.updateConfigArtifacts(props.configArtifacts.concat([{}]));
```

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx (hunk 1 above)
Comment (Jammy-Louie) [reply app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]: 👍

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx (hunk 2 above)
Comment (Jammy-Louie) [reply app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]: 👍

File: app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx (hunk 4 above)
Comment (Jammy-Louie) [reply app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx]: 👍

**Comment by Jammy-Louie** [comment]
@maggieneterval Could never ignore such sound and wise words

**Pull request merged**